test/test-listeners.js

```diff
@@ -1,13 +1,26 @@
 import XElement from '../x-element.js';
 import { assert, it } from '../../../@netflix/x-test/x-test.js';
 
+class TestElementChild extends HTMLElement {
+  connectedCallback() {
```

Comment:
Here's the test setup (note that this extends a basic `HTMLElement` by the way) — we have an element that emits an event as soon as it's connected to the DOM. Maybe not the _best_ pattern, but honestly, not completely pathological.

Seems fairly clear that we should try and support this.